### PR TITLE
[zh-cn] sync 2022-12-19-devicemanager-ga.md

### PR DESCRIPTION
**Pull request by qlijin**
Sync file 2022-12-19-devicemanager-ga.md

```
/content/en/blog/_posts/2022-12-19-devicemanager-ga.md
```

### CODE REVIEW

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+<!--
+Within the kubelet, the Device Manager facilitates communication with device plugins
+using gRPC through Unix sockets. Device Manager and Device plugins both act as gRPC
+servers and clients by serving and connecting to the exposed gRPC services respectively.
+Device plugins serve a gRPC service that kubelet connects to for device discovery,
+advertisement (as extended resources) and allocation. Device Manager connects to
+the `Registration` gRPC service served by kubelet to register itself with kubelet.
+-->
+在 kubelet 中，设备管理器通过 Unix 套接字使用 gRPC 实现与设备插件的通信。
+设备管理器和设备插件都充当 gRPC 服务器和客户端的角色，分别为暴露的 gRPC 服务提供服务和进行连接。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:43]:
```suggestion
设备管理器和设备插件都充当 gRPC 服务器和客户端的角色，分别提供暴露的 gRPC 服务并进行连接。
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+<!--
+### Kubelet APIs moved to kubelet staging repo
+External facing `deviceplugin` API packages moved from `k8s.io/kubernetes/pkg/kubelet/apis/`
+to `k8s.io/kubelet/pkg/apis/` in v1.17. Refer to [Move external facing kubelet apis to staging](https://github.com/kubernetes/kubernetes/pull/83551) for more details on the rationale behind this change.
+-->
+### Kubelet APIs 移至 kubelet 暂存库
+在 v1.17 版本中，面向外部的 `deviceplugin` API 包已从 `k8s.io/kubernetes/pkg/kubelet/apis/`
+移动到了 `k8s.io/kubelet/pkg/apis/`。有关此变更背后的更多详细信息，
+请参阅[移动面向外部的 kubelet 接口到暂存区](https://github.com/kubernetes/kubernetes/pull/83551)
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:79]:
```suggestion
请参阅 [Move external facing kubelet apis to staging](https://github.com/kubernetes/kubernetes/pull/83551)
```
这种可以不译，跟跳转后的 PR title 保持一致

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+<!--
+Additional gRPC endpoints introduced:
+  1. `GetDevicePluginOptions` is used by device plugins to communicate
+     options to the `DeviceManager` in order to indicate if `PreStartContainer`,
+     `GetPreferredAllocation` or other future optional calls are supported and
+     can be called before making devices available to the container.
+-->
+新增了额外的 gRPC 端点：
+  1. `GetDevicePluginOptions` 用于设备插件向 `DeviceManager` 传递选项，以指示是否支持
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:94]:
```suggestion

1. `GetDevicePluginOptions` 用于设备插件向 `DeviceManager` 传递选项，以指示是否支持
```
这种就不需要 follow en 了，保持正确的缩进就好

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+     `PreStartContainer`、`GetPreferredAllocation` 或其他将来的可选调用，
+     并可在向容器提供设备之前进行调用。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:96]:
```suggestion
   `PreStartContainer`、`GetPreferredAllocation` 或其他将来的可选调用，
   并可在向容器提供设备之前进行调用。
```
ditto

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+     into its allocation decisions. The `DeviceManager` will call out to a
+     plugin at pod admission time asking for a preferred device allocation
+     of a given size from a list of available devices to make a more informed
+     decision. E.g. Specifying inter-device constraints to indicate preferrence
+     on best-connected set of devices when allocating devices to a container.
+-->
+  2. `GetPreferredAllocation` 允许设备插件将优先分配信息传递给 `DeviceManager`，
+     使其能够将此信息纳入其分配决策中。 `DeviceManager` 在 pod
+     准入时向插件请求指定大小的优选设备分配， 以便做出更明智的决策。
+     例如，在为容器分配设备时，指定设备间的约束条件以表明对最佳连接设备集合的偏好。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:110]:
```suggestion
2. `GetPreferredAllocation` 允许设备插件将优先分配信息传递给 `DeviceManager`，
   使其能够将此信息纳入其分配决策中。`DeviceManager` 在 Pod
   准入时向插件请求指定大小的优选设备分配，以便做出更明智的决策。
   例如，在为容器分配设备时，指定设备间的约束条件以表明对最佳连接设备集合的偏好。
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+<!--
+  1. `PreStartContainer` is called before each container start if indicated by
+     device plugins during registration phase. It allows Device Plugins to run device
+     specific operations on the Devices requested. E.g. reconfiguring or
+     reprogramming FPGAs before the container starts running. 
+-->
+  3. 在注册阶段由设备插件指示时，`PreStartContainer` 会在每次容器启动之前被调用。
+     它允许设备插件在所请求的设备上执行特定的设备操作。
+     例如，在容器启动前对 FPGA 进行重新配置或重新编程。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:120]:
```suggestion
3. 在注册阶段由设备插件指示时，`PreStartContainer` 会在每次容器启动之前被调用。
   它允许设备插件在所请求的设备上执行特定的设备操作。
   例如，在容器启动前对 FPGA 进行重新配置或重新编程。
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+<!--
+Pull Requests that introduced these changes are here:
+1. [Invoke preStart RPC call before container start, if desired by plugin](https://github.com/kubernetes/kubernetes/pull/58282)
+1. [Add GetPreferredAllocation() call to the v1beta1 device plugin API](https://github.com/kubernetes/kubernetes/pull/92665)
+-->
+引入这些更改的拉取请求在这里：
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:127]:
```suggestion
引入这些更改的 PR 为：
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+1. [如果插件需要，在容器启动之前调用 preStart RPC](https://github.com/kubernetes/kubernetes/pull/58282)
+1. [向 v1beta1 版本的设备插件 API 添加 GetPreferredAllocation() 调用](https://github.com/kubernetes/kubernetes/pull/92665)
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:129]:
```suggestion
1. [Invoke preStart RPC call before container start, if desired by plugin](https://github.com/kubernetes/kubernetes/pull/58282)
1. [Add GetPreferredAllocation() call to the v1beta1 device plugin API](https://github.com/kubernetes/kubernetes/pull/92665)
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+<!--
+With introduction of the above endpoints the interaction between Device Manager in
+kubelet and Device Manager can be shown as below:
+-->
+引入上述端点后，kubelet中的设备管理器与设备管理器之间的交互可以如下展示：
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:135]:
```suggestion
引入上述端点后，kubelet 中的设备管理器与设备管理器之间的交互如下所示：
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+<!--
+### Dynamic resource allocation
+In Kubernetes 1.26, inspired by how [Persistent Volumes](/docs/concepts/storage/persistent-volumes)
+are handled in Kubernetes, [Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
+has been introduced to cater to devices that have more sophisticated resource requirements like:
+-->
+### 动态资源分配
+在 Kubernetes 1.26 中，受 Kubernetes 处理
+[持久卷(Persistent Volumes)](/zh-cn/docs/concepts/storage/persistent-volumes) 方式的启发，
+引入了 [动态资源分配](/zh-cn/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)，
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:176]:
```suggestion

在 Kubernetes 1.26 中，受 Kubernetes
处理[持久卷](/zh-cn/docs/concepts/storage/persistent-volumes)方式的启发，
引入了[动态资源分配](/zh-cn/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)，
```

File: content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+不，设备插件 API 仍然不稳定；目前最新的可用设备插件 API 版本是 `v1beta1`。
+社区计划引入 `v1beta2` API，以便同时为多个插件 API 提供服务。
+对每个 API 的调用都具有请求/响应类型，可以在不明确升级 API 的情况下添加对新 API 版本的支持。
+
+<!--
+In addition to that, there are existing proposals in the community to introduce additional
+endpoints [KEP-3162: Add Deallocate and PostStopContainer to Device Manager API](https://github.com/kubernetes/enhancements/issues/3162).
+-->
+除此之外，社区中存在一些提案，打算引入额外的端点 [KEP-3162: 将 Deallocate 和 PostStopContainer
+添加到设备管理器 API](https://github.com/kubernetes/enhancements/issues/3162)。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-12-19-devicemanager-ga.md/index.md:209]:
```suggestion
除此之外，社区中存在一些提案，打算引入额外的端点
[KEP-3162: Add Deallocate and PostStopContainer to Device Manager API](https://github.com/kubernetes/enhancements/issues/3162)。
```

**Comment by windsonsea** [comment]
You need to squash your commits into a single one. Refer to:
https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits

**Comment by qlijin** [comment]
> You need to squash your commits into a single one. Refer to: https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits

OK

**Review by windsonsea — COMMENTED** [review]
/approve

**Pull request merged**